sale_order_revision/models/sale_order.py

```diff
@@ -51,21 +33,6 @@ class sale_order(models.Model):
          'Order Reference and revision must be unique per Company.'),
     ]
 
-    @api.multi
-    def copy_quotation(self):
-        self.ensure_one()
-        revision_self = self.with_context(new_sale_revision=True)
-        action = super(sale_order, revision_self).copy_quotation()
-        old_revision = self.browse(action['res_id'])
-        action['res_id'] = self.id
-        self.delete_workflow()
-        self.create_workflow()
-        self.write({'state': 'draft'})
-        msg = _('New revision created: %s') % self.name
-        self.message_post(body=msg)
-        old_revision.message_post(body=msg)
-        return action
-
     @api.returns('self', lambda value: value.id)
     @api.multi
     def copy(self, defaults=None):
```

Comment:
I haven't used this module before but it doesn't work as expected from reading the doc.
This copy method modifies the original sale.order to change the name, and the copy is set as cancelled and inactive.
From the description, IMO the original sale.order should be set to inactive (it's already in state cancelled) and the copied sale.order should be modified with new revision number, state draft and set as active. Otherwise the metadata from both sale.order are irrelevant.